linker-common-lib/src/main/java/com/linker/common/network/TcpSocketClient.java

```diff
@@ -0,0 +1,117 @@
+package com.linker.common.network;
+
+import com.linker.common.MessageContent;
+import com.linker.common.MessageContentOutput;
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.*;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.codec.LengthFieldBasedFrameDecoder;
+import io.netty.handler.codec.LengthFieldPrepender;
+import io.netty.handler.codec.bytes.ByteArrayDecoder;
+import io.netty.handler.codec.bytes.ByteArrayEncoder;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+@Slf4j
+public class TcpSocketClient implements SocketClient {
```

Comment:
`@Component`?

Reply:
This is like a WebSocket in javascript.  When it's used, we need to defined a Service to wrap it, and  this will delivered to customer who might not use spring in the future